### PR TITLE
Add descriptions to schemas by hand to MM resources

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/8236

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:deprecation
pubsub: deprecated `path` field in `google_pubsub_subscription` resource as it is identical value to `id` field.
```
```release-note:deprecation
kms: deprecated `self_link` field in `google_kms_keyring` and `google_kms_cryptokey` resource as it is identical value to `id` field.
```
```release-note:deprecation
compute: deprecated `source_disk_url` field in `google_compute_snapshot`.
```

### CODE REVIEW

**Comment by upodroid** [comment]
I looked at all the schema entries manually loaded in MM and I have a few questions:

- `source_disk_link` in snapshot.erb(google_compute_snapshot). This is not documented in the provider and all it seems to be doing is  converting selflinks to a one with v1 in the url.
- KMS self_link also needs deprecating as it is identical to id and KMS API doesn't have self_links.

**Review by rileykarson — APPROVED** [review]
LGTM. I'm going to hold merging in case you want to deal with either of the other two fields at the same time:

* source_disk_link appears to have outputted the source_disk value from the API at some point (and source_disk did not), but is redundant with source_disk now and should be safe to deprecate. https://github.com/hashicorp/terraform-provider-google/commit/50e8c65740917a7f40f6b4f3c66f787bc47f92b8

* The KMS self_link field came from a time when we added artificial self_link fields to non-GCE resources. If it's redundant with id, that should be fine to deprecate.

**Comment by upodroid** [comment]
I'll deprecate both and open issues to remove them

File: templates/terraform/extra_schema_entry/kms_self_link.erb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 "self_link": &schema.Schema{
 	Type:     schema.TypeString,
 	Computed: true,
+	Deprecated: "Deprecated in favor of id. This field will be removed in the next release of the provider. ",
```
Comment (rileykarson) [inline templates/terraform/extra_schema_entry/kms_self_link.erb:4]:
```suggestion
	Deprecated: "Deprecated in favor of id, which contains an identical value. This field will be removed in the next major release of the provider.",
```

File: templates/terraform/extra_schema_entry/pubsub_subscription_path.erb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 "path": &schema.Schema{
 	Type:     schema.TypeString,
 	Computed: true,
+	Deprecated:  "Deprecated in favor of id. This field will be removed in the next major release of the provider.",
```
Comment (rileykarson) [inline templates/terraform/extra_schema_entry/pubsub_subscription_path.erb:4]:
```suggestion
	Deprecated:  "Deprecated in favor of id, which contains an identical value. This field will be removed in the next major release of the provider.",
```

File: templates/terraform/extra_schema_entry/snapshot.erb [hunk 3]
```diff
@@ -15,4 +15,5 @@
 "source_disk_link": {
 	Type:       schema.TypeString,
 	Computed:   true,
+	Deprecated: "Deprecated in favor of source_disk. This field will be removed in the next major release of the provider.",
```
Comment (rileykarson) [inline templates/terraform/extra_schema_entry/snapshot.erb:18]:
```suggestion
	Deprecated: "Deprecated in favor of source_disk, which contains a compatible value. This field will be removed in the next major release of the provider.",
```

**Review by rileykarson — COMMENTED** [review]
Added some minor suggestions on the deprecation notes. Can you also mark them as deprecated in the website documentation as well?

File: templates/terraform/extra_schema_entry/snapshot.erb (hunk 3 above)
Comment (upodroid) [reply templates/terraform/extra_schema_entry/snapshot.erb]: fixed

**Comment by upodroid** [comment]
bump

**Pull request merged**